transaction-spec/src/main/java/com/bloxbean/cardano/client/transaction/spec/Value.java

```diff
@@ -150,19 +149,57 @@ private Map serializeMultiAssets() {
      * @param that parameter to sum with
      * @return {@link Value} of the Sum
      */
-    public Value plus(Value that) {
+    public Value add(Value that) {
         BigInteger sumCoin = (getCoin() == null ? BigInteger.ZERO.add(that.getCoin()) : getCoin().add(that.getCoin()));
         List<MultiAsset> sumMultiAssets = MultiAsset.mergeMultiAssetLists(getMultiAssets(), that.getMultiAssets());
         return Value.builder().coin(sumCoin).multiAssets(sumMultiAssets).build();
     }
 
+    /**
+     * Use add(Value that) instead
+     *
+     * @param that
+     * @return
+     */
+    @Deprecated
+    public Value plus(Value that) {
+        return this.add(that);
+    }
+
+    public Value add(String policyId, String assetName, BigInteger amount) {
+        return this.add(from(policyId, assetName, amount));
+    }
+
+    public Value add(String unit, BigInteger amount) {
+        Tuple<String, String> policyAndAssetName = getPolicyAndAssetName(unit);
+        return this.add(from(policyAndAssetName._1, policyAndAssetName._2, amount));
+    }
+
+
+    public static Value from(String policyId, String assetName, BigInteger amount) {
```

Comment:
@nemo83 Thanks for the PR.

1. `from(policyId, assetName, amount)`
2. `fromLovelace(amount)`

If I am not wrong, in the case of lovelace, the policy ID is empty. But, I need to verify that. To avoid confusion, as these values are represented differently in the Value structure, we can keep #1 for native assets and #2 for lovelace. For #1, we can add a null check for policy id. Null check for asset name is not required as it could be empty.

Also, we can add an overloaded method 
`addLovelace(amount)`.

And in `Value add(String unit, BigInteger amount)` , check 

- if `unit=lovelace` then addLovelace(amount)
- else `add(from(...)` (current code)

Reply:
Thanks for the review.

For lovelace values, both policy id and asset name should be empty byte array.  

The class `Amount` in CCL has just `unit` attribute which is set to `lovelace` for ada amount, which confused me in the past coz when doing substring, it would then be policyId = lovelace.  The `Amt` class in yaci has policyId empty and asset name equals to lovelace.
IMHO, both should be empty, setting either of them or unit to lovelace just complicates things. Hence the policy check and asset name check.

Will leverage `AssetUtil` to try to simplify and add the other `add` methods